selfdrive/car/honda/values.py

```diff
@@ -838,6 +839,29 @@ class CAR:
       b'77959-T5R-A230\x00\x00',
     ],
   },
+  CAR.FREED: {
+    (Ecu.gateway, 0x18daeff1, None): [
+      b'38897-TDK-J010\x00\x00',
+    ],
+    (Ecu.eps, 0x18da30f1, None): [
+      b'39990-TDK-J050\x00\x00',
+      b'39990-TDK-N020\x00\x00',
+    ],
+    # TODO: vsa is "essential" for fpv2 but doesn't appear on some models
+    (Ecu.vsa, 0x18da28f1, None): [
+      b'57114-TDK-J120\x00\x00',
+      b'57114-TDK-J330\x00\x00',
+    ],
```

Comment:
are there other options for this vsa fingerprint issue than 1) add another CAR entry, 2) rely on fuzzy match?

Reply:
we should look into why that is, but there's already precedent for exceptions like this https://github.com/commaai/openpilot/blob/master/selfdrive/car/fw_versions.py#L224

Reply:
(is it correct that "vsa" is in the essential set?)

Reply:
so perhaps this?

```python
      # vsa is missing on some Honda Freed sub-models.  (Does this ECU need to be essential?)
      if ecu_type == Ecu.vsa and candidate == HONDA.FREED and found_version is None:
        continue
```

Reply:
yeah, something like that, but I'd like to understand why this happens first

Reply:
`VSA_STATUS` is active in the route, so this implies it's not coming from the vsa ECU

![Screen Shot 2021-09-25 at 9 29 17 AM](https://user-images.githubusercontent.com/1708631/134751711-d7ebc9b4-0c64-4be1-a672-ee8db55c54ff.png)

2c4292a5cd10536c|2021-09-24--22-21-53

Reply:
Oh never mind-- in this route the vsa actually appeared in carFw.

My first two engine starts didn't have vsa, then I removed the entry from the fingerprint, and it appeared in canFw on subsequent engine starts 🤷